core/src/test/java/org/apache/iceberg/TestSnapshotManager.java

```diff
@@ -700,81 +689,82 @@ public void testAttemptToRollbackToCurrentSnapshot() {
     table.manageSnapshots().rollbackTo(currentSnapshotId).commit();
   }
 
-  @Test
+  @TestTemplate
   public void testSnapshotManagerThroughTransaction() {
     table.newAppend().appendFile(FILE_A).commit();
     Snapshot snapshotAfterFirstAppend = readMetadata().currentSnapshot();
     validateSnapshot(null, snapshotAfterFirstAppend, FILE_A);
 
     table.newAppend().appendFile(FILE_B).commit();
     validateSnapshot(snapshotAfterFirstAppend, readMetadata().currentSnapshot(), FILE_B);
-    Assert.assertEquals("Table should be on version 2 after appending twice", 2, (int) version());
+    assertThat(version()).as("Table should be on version 2 after appending twice").isEqualTo(2);
 
     TableMetadata base = readMetadata();
     Transaction txn = table.newTransaction();
 
-    Assert.assertSame(
-        "Base metadata should not change when transaction is created", base, readMetadata());
-    Assert.assertEquals(
-        "Table should be on version 2 after creating transaction", 2, (int) version());
+    assertThat(readMetadata())
+        .as("Base metadata should not change when transaction is created")
+        .isEqualTo(base);
+    assertThat(version())
+        .as("Table should be on version 2 after creating transaction")
+        .isEqualTo(2);
 
     ManageSnapshots manageSnapshots = txn.manageSnapshots();
-    Assert.assertNotNull(manageSnapshots);
+    assertThat(manageSnapshots).isNotNull();
 
-    Assert.assertSame(
-        "Base metadata should not change when manageSnapshots is created", base, readMetadata());
-    Assert.assertEquals(
-        "Table should be on version 2 after creating manageSnapshots", 2, (int) version());
+    assertThat(readMetadata())
+        .as("Base metadata should not change when manageSnapshots is created")
+        .isEqualTo(base);
```

Comment:
this should be `isSameAs()` as this check actually verifies that both objects are referentially equal. Same for the other places that previously used `assertSame()`

Reply:
Okay sure. Thanks for the advice.